### PR TITLE
Update to scala 2.13

### PR DESCRIPTION
**Pull request by kennydaniel**
Bump dependency versions and fix warnings and tests. Cross builds to 2.11, 2.12, 2.13.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -3,20 +3,20 @@ organization := "com.algorithmia"
 
 version := "0.9.6-SNAPSHOT"
 
-scalaVersion := "2.11.12"
+scalaVersion := "2.13.0"
```
Comment (dinoboy197) [inline build.sbt:6]: I believe when you do cross-building, you don't need to specify this line?

**Comment by dinoboy197** [comment]
Are all of the tests passing locally for this? I don't see this as a CI build in Jenkins or Puppet.

File: build.sbt (hunk 1 above)
Comment (kennydaniel) [reply build.sbt:6]: This line determines the "default" scala version when doing things like running `sbt test` locally.

**Comment by kennydaniel** [comment]
Tests are passing but yes I agree it should be added to CI, I'll take a look

File: build.sbt (hunk 1 above)
Comment (dinoboy197) [reply build.sbt:6]: If you want to use Jenkins to automate tests across all versions (so each supported Scala version gets tested), here is how we do it in Logpig:

* build.sbt - https://github.com/algorithmiaio/logpig/blob/master/build.sbt
* Jenkinsfile - https://github.com/algorithmiaio/logpig/blob/master/Jenkinsfile

**Review by jjbooth74 — CHANGES_REQUESTED** [review]
:+1: for improved error messaging

When I pulled and ran tests locally, I get an error (see below).  I agree with Taylor this repo would benefit from CI covering multiple versions for this repo, but that's beyond the scope of this PR
```
$ sbt test
[info] Loading settings for project global-plugins from metals.sbt ...
[info] Loading global plugins from /home/josh/.sbt/1.0/plugins
[info] Loading project definition from /home/josh/Development/algorithmia-scala/project
[info] Loading settings for project algorithmia-scala from build.sbt,sonatype.sbt ...
[info] Set current project to algorithmia-scala (in build file:/home/josh/Development/algorithmia-scala/)
[info] AlgorithmTest
[info] AlgoResponseTest
[info] Total for specification AlgorithmTest
[info] Finished in 1 second, 787 ms
[info] 0 example, 0 failure, 0 error
[info] Total for specification AlgoResponseTest
[info] Finished in 1 second, 787 ms
[info] 0 example, 0 failure, 0 error
Exception in thread "specs2-1" java.lang.OutOfMemoryError: Java heap space
	at java.util.Arrays.copyOf(Arrays.java:3236)
	at java.io.ByteArrayOutputStream.grow(ByteArrayOutputStream.java:118)
	at java.io.ByteArrayOutputStream.ensureCapacity(ByteArrayOutputStream.java:93)
	at java.io.ByteArrayOutputStream.write(ByteArrayOutputStream.java:153)
	at sun.net.www.http.PosterOutputStream.write(PosterOutputStream.java:78)
	at com.algorithmia.client.HttpClient.copy(HttpClient.scala:93)
	at com.algorithmia.client.HttpClient.$anonfun$put$1(HttpClient.scala:62)
	at com.algorithmia.client.HttpClient.$anonfun$put$1$adapted(HttpClient.scala:58)
	at com.algorithmia.client.HttpClient$$Lambda$4665/1978425136.apply(Unknown Source)
	at scalaj.http.HttpRequest.$anonfun$doConnection$4(Http.scala:361)
	at scalaj.http.HttpRequest.$anonfun$doConnection$4$adapted(Http.scala:361)
	at scalaj.http.HttpRequest$$Lambda$4489/782052001.apply(Unknown Source)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:576)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:574)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:904)
	at scalaj.http.HttpRequest.doConnection(Http.scala:361)
	at scalaj.http.HttpRequest.exec(Http.scala:343)
	at scalaj.http.HttpRequest.asString(Http.scala:492)
	at com.algorithmia.client.HttpClient.put(HttpClient.scala:64)
	at com.algorithmia.data.DataFile.put(DataFile.scala:46)
	at DataFileGenericTest.$anonfun$new$35(DataFileGenericTest.scala:95)
	at DataFileGenericTest$$Lambda$3807/1537082042.apply(Unknown Source)
	at org.specs2.matcher.MatchResult$$anon$12.$anonfun$asResult$1(MatchResult.scala:344)
	at org.specs2.matcher.MatchResult$$anon$12$$Lambda$4127/43140322.apply(Unknown Source)
	at org.specs2.execute.ResultExecution.execute(ResultExecution.scala:22)
	at org.specs2.execute.ResultExecution.execute$(ResultExecution.scala:21)
	at org.specs2.execute.ResultExecution$.execute(ResultExecution.scala:123)
	at org.specs2.execute.Result$$anon$4.asResult(Result.scala:246)
	at org.specs2.execute.AsResult$.apply(AsResult.scala:32)
	at org.specs2.matcher.MatchResult$$anon$12.asResult(MatchResult.scala:344)
	at org.specs2.execute.AsResult$.apply(AsResult.scala:32)
	at org.specs2.specification.core.AsExecution$$anon$1.$anonfun$execute$1(AsExecution.scala:17)
```

**Comment by kennydaniel** [comment]
@jjbooth74 I was able to repro your OOM on a different computer. I just changed the code to use chunked encoding when streaming a large file. When you get a chance can you pull and try again?

**Comment by jjbooth74** [comment]
@kennydaniel i was able to get past the OOM but i'm still getting test failures locally.  Looks like some things are permissions issues, are there pre-requisites i need to set up for tests to run successfully?

```
$ sbt test
[info] Loading settings for project global-plugins from metals.sbt ...
[info] Loading global plugins from /home/josh/.sbt/1.0/plugins
[info] Loading project definition from /home/josh/Development/algorithmia-scala/project
[info] Loading settings for project algorithmia-scala from build.sbt,sonatype.sbt ...
[info] Set current project to algorithmia-scala (in build file:/home/josh/Development/algorithmia-scala/)
[info] Compiling 1 Scala source to /home/josh/Development/algorithmia-scala/target/scala-2.13/classes ...
[info] Done compiling.
[info] AlgoResponseTest
[info] AlgorithmTest
[info] Total for specification AlgoResponseTest
[info] Total for specification AlgorithmTest
[info] Finished in 1 second, 889 ms
[info] 0 example, 0 failure, 0 error
[info] Finished in 1 second, 889 ms
[info] 0 example, 0 failure, 0 error
[info] DataFileTest
[info] api key should
[error]   x define environment variable ALGORITHMIA_API_KEY
[error]    null is null (DataFileGenericTest.scala:13)
[info] test file should
[info]   + have right size
[info] path operations should
[info]   + get filename
[info]   + get parent
[info] put operations should
[error]   x create file
[error]    the value is false (DataFileGenericTest.scala:55)
[error]   x string upload
[error]    the value is false (DataFileGenericTest.scala:67)
[error]   x file upload
[error]    the value is false (DataFileGenericTest.scala:81)
[error]   ! large file upload
[error]    java.io.IOException: Error writing request body to server (HttpURLConnection.java:3587)
[error] sun.net.www.protocol.http.HttpURLConnection$StreamingOutputStream.checkError(HttpURLConnection.java:3587)
[error] sun.net.www.protocol.http.HttpURLConnection$StreamingOutputStream.write(HttpURLConnection.java:3570)
[error] com.algorithmia.client.HttpClient.copy(HttpClient.scala:96)
[error] com.algorithmia.client.HttpClient.$anonfun$put$1(HttpClient.scala:65)
[error] com.algorithmia.client.HttpClient.$anonfun$put$1$adapted(HttpClient.scala:58)
[error] scalaj.http.HttpRequest.$anonfun$doConnection$4(Http.scala:361)
[error] scalaj.http.HttpRequest.$anonfun$doConnection$4$adapted(Http.scala:361)
[error] scalaj.http.HttpRequest.doConnection(Http.scala:361)
[error] scalaj.http.HttpRequest.exec(Http.scala:343)
[error] scalaj.http.HttpRequest.asString(Http.scala:492)
[error] com.algorithmia.client.HttpClient.put(HttpClient.scala:67)
[error] com.algorithmia.data.DataFile.put(DataFile.scala:42)
[error] DataFileGenericTest.$anonfun$new$32(DataFileGenericTest.scala:88)
[error]   ! large file upload stream
[error]    java.io.IOException: Error writing request body to server (HttpURLConnection.java:3587)
[error] sun.net.www.protocol.http.HttpURLConnection$StreamingOutputStream.checkError(HttpURLConnection.java:3587)
[error] sun.net.www.protocol.http.HttpURLConnection$StreamingOutputStream.write(HttpURLConnection.java:3570)
[error] com.algorithmia.client.HttpClient.copy(HttpClient.scala:96)
[error] com.algorithmia.client.HttpClient.$anonfun$put$1(HttpClient.scala:65)
[error] com.algorithmia.client.HttpClient.$anonfun$put$1$adapted(HttpClient.scala:58)
[error] scalaj.http.HttpRequest.$anonfun$doConnection$4(Http.scala:361)
[error] scalaj.http.HttpRequest.$anonfun$doConnection$4$adapted(Http.scala:361)
[error] scalaj.http.HttpRequest.doConnection(Http.scala:361)
[error] scalaj.http.HttpRequest.exec(Http.scala:343)
[error] scalaj.http.HttpRequest.asString(Http.scala:492)
[error] com.algorithmia.client.HttpClient.put(HttpClient.scala:67)
[error] com.algorithmia.data.DataFile.put(DataFile.scala:46)
[error] DataFileGenericTest.$anonfun$new$35(DataFileGenericTest.scala:95)
[info] get operations should
[error]   x get string
[error]    the value is false (DataFileGenericTest.scala:108)
[error]   ! get large file
[error]    java.io.IOException: Error writing request body to server (HttpURLConnection.java:3587)
[error] sun.net.www.protocol.http.HttpURLConnection$StreamingOutputStream.checkError(HttpURLConnection.java:3587)
[error] sun.net.www.protocol.http.HttpURLConnection$StreamingOutputStream.write(HttpURLConnection.java:3570)
[error] com.algorithmia.client.HttpClient.copy(HttpClient.scala:96)
[error] com.algorithmia.client.HttpClient.$anonfun$put$1(HttpClient.scala:65)
[error] com.algorithmia.client.HttpClient.$anonfun$put$1$adapted(HttpClient.scala:58)
[error] scalaj.http.HttpRequest.$anonfun$doConnection$4(Http.scala:361)
[error] scalaj.http.HttpRequest.$anonfun$doConnection$4$adapted(Http.scala:361)
[error] scalaj.http.HttpRequest.doConnection(Http.scala:361)
[error] scalaj.http.HttpRequest.exec(Http.scala:343)
[error] scalaj.http.HttpRequest.asString(Http.scala:492)
[error] com.algorithmia.client.HttpClient.put(HttpClient.scala:67)
[error] com.algorithmia.data.DataFile.put(DataFile.scala:42)
[error] DataFileGenericTest.$anonfun$new$44(DataFileGenericTest.scala:123)
[info] file types should
[info]   + files are files
[info] Total for specification DataFileTest
[info] Finished in 4 seconds, 396 ms
[info] 12 examples, 5 failures, 3 errors
[info] DataDirectoryTest
[info] api key should
[error]   x define environment variable ALGORITHMIA_API_KEY
[error]    null is null (DataDirectoryGenericTest.scala:11)
[info] path operations should
[info]   + get dirname
[info]   + get parent
[info] create operations should
[error]   x create directory
[error]    the value is false (DataDirectoryGenericTest.scala:42)
[info] list operations should
[error]   ! list directory
[error]    java.util.NoSuchElementException: Failed to fetch listing page, status code: 400 (AbstractDataIterator.scala:46)
[error] com.algorithmia.data.AbstractDataIterator.attemptToLoadFirstPage(AbstractDataIterator.scala:46)
[error] com.algorithmia.data.AbstractDataIterator.hasNext(AbstractDataIterator.scala:17)
[error] com.algorithmia.data.DataDirectory$$anon$1.foreach(DataDirectory.scala:30)
[error] DataDirectoryGenericTest.$anonfun$new$19(DataDirectoryGenericTest.scala:59)
[error]   ! list with trailing slash
[error]    java.util.NoSuchElementException: Failed to fetch listing page, status code: 400 (AbstractDataIterator.scala:46)
[error] com.algorithmia.data.AbstractDataIterator.attemptToLoadFirstPage(AbstractDataIterator.scala:46)
[error] com.algorithmia.data.AbstractDataIterator.hasNext(AbstractDataIterator.scala:17)
[error] com.algorithmia.data.DataDirectory$$anon$1.foreach(DataDirectory.scala:30)
[error] DataDirectoryGenericTest.dataDirListIterable(DataDirectoryGenericTest.scala:143)
[error] DataDirectoryGenericTest.$anonfun$new$29(DataDirectoryGenericTest.scala:70)
[error]   ! list without trailing slash
[error]    java.util.NoSuchElementException: Failed to fetch listing page, status code: 400 (AbstractDataIterator.scala:46)
[error] com.algorithmia.data.AbstractDataIterator.attemptToLoadFirstPage(AbstractDataIterator.scala:46)
[error] com.algorithmia.data.AbstractDataIterator.hasNext(AbstractDataIterator.scala:17)
[error] com.algorithmia.data.DataDirectory$$anon$1.foreach(DataDirectory.scala:30)
[error] DataDirectoryGenericTest.dataDirListIterable(DataDirectoryGenericTest.scala:143)
[error] DataDirectoryGenericTest.$anonfun$new$30(DataDirectoryGenericTest.scala:74)
[error]   ! list large directory
[error]    java.util.NoSuchElementException: Failed to fetch listing page, status code: 400 (AbstractDataIterator.scala:46)
[error] com.algorithmia.data.AbstractDataIterator.attemptToLoadFirstPage(AbstractDataIterator.scala:46)
[error] com.algorithmia.data.AbstractDataIterator.hasNext(AbstractDataIterator.scala:17)
[error] com.algorithmia.data.DataDirectory$$anon$1.foreach(DataDirectory.scala:30)
[error] DataDirectoryGenericTest.$anonfun$new$31(DataDirectoryGenericTest.scala:93)
[error]   ! list empty directory
[error]    java.util.NoSuchElementException: Failed to fetch listing page, status code: 400 (AbstractDataIterator.scala:46)
[error] com.algorithmia.data.AbstractDataIterator.attemptToLoadFirstPage(AbstractDataIterator.scala:46)
[error] com.algorithmia.data.AbstractDataIterator.hasNext(AbstractDataIterator.scala:17)
[error] com.algorithmia.data.DataDirectory$$anon$2.foreach(DataDirectory.scala:36)
[error] DataDirectoryGenericTest.$anonfun$new$38(DataDirectoryGenericTest.scala:113)
[info] file types should
[info]   + directories are directories
[info] directory permissions should
[error]   ! get directory permissions
[error]    java.io.IOException: Failed to get permissions, status code 400 (DataDirectory.scala:64)
[error] com.algorithmia.data.DataDirectory.getPermissions(DataDirectory.scala:64)
[error] DataDirectoryTest.$anonfun$new$3(DataDirectoryTest.scala:17)
[error] DataDirectoryTest.$anonfun$new$2(DataDirectoryTest.scala:17)
[error]   ! create directory with permissions
[error]    java.io.IOException: Failed to get permissions, status code 400 (DataDirectory.scala:64)
[error] com.algorithmia.data.DataDirectory.getPermissions(DataDirectory.scala:64)
[error] DataDirectoryTest.$anonfun$new$6(DataDirectoryTest.scala:24)
[error] DataDirectoryTest.$anonfun$new$5(DataDirectoryTest.scala:24)
[error]   ! update directory permissions
[error]    java.io.IOException: Failed to get permissions, status code 400 (DataDirectory.scala:64)
[error] com.algorithmia.data.DataDirectory.getPermissions(DataDirectory.scala:64)
[error] DataDirectoryTest.$anonfun$new$9(DataDirectoryTest.scala:31)
[error] DataDirectoryTest.$anonfun$new$8(DataDirectoryTest.scala:31)
[info] Total for specification DataDirectoryTest
[info] Finished in 1 minute 30 seconds, 634 ms
[info] 13 examples, 2 failures, 8 errors
[error] Error: Total 25, Failed 7, Errors 11, Passed 7
[error] Error during tests:
[error] 	DataDirectoryTest
[error] 	DataFileTest
[error] (Test / test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 102 s, completed Jul 12, 2019 12:02:42 PM
```

**Comment by kennydaniel** [comment]
You need to set an environment variable with your api key `export ALGORITHMIA_API_KEY=...`

**Review by jjbooth74 — APPROVED** [review]
Was able to get tests to pass :+1:

**Pull request merged**